src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2889 @@
+    REAL(r_kind) :: var9,var10,var11,var12
+
+    assign_para_dsd_TM%T_qr = var1
+    assign_para_dsd_TM%T_qs = var2
+    assign_para_dsd_TM%T_qh = var3
+    assign_para_dsd_TM%T_qg = var4
+    assign_para_dsd_TM%T_Ntr = var5
+    assign_para_dsd_TM%T_Nts = var6
+    assign_para_dsd_TM%T_Nth = var7
+    assign_para_dsd_TM%T_Ntg = var8
+    assign_para_dsd_TM%T_alfr = var9
+    assign_para_dsd_TM%T_alfs = var10
+    assign_para_dsd_TM%T_alfh = var11
+    assign_para_dsd_TM%T_alfg = var12
+  END FUNCTION assign_para_dsd_TM
+
+
+TYPE(T_obs_dual) FUNCTION rainIceRefl(var_dsd,rho,flg)
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! This subroutine calculates the partial reflectivity factor
+! of melting(wet) snow/hail at horizontal polarization
+! and compute total reflectivity as a sum of those.
+! The same formula used in shfactor is used with different
+! alpha and beta coefficients that contain the effect of the fraction
+! of water in the melting snow to take the melting layer into account.
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/29/2007
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  use radarz_cst, only: mphyopt, MFflg
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! External Function declaration
+!-----------------------------------------------------------------------
+
+!  REAL(r_kind), EXTERNAL    :: snow_alpha_a, hail_alpha_a, grpl_alpha_a
+!  REAL(r_kind), EXTERNAL    :: snow_alpha_b, hail_alpha_b, grpl_alpha_b
+!  INTEGER(i_kind), EXTERNAL :: get_qgh_opt
+
+!-----------------------------------------------------------------------
+! Declare local variables.
+!-----------------------------------------------------------------------
+ 
+  TYPE(T_para_dsd) :: var_dsd
+  REAL(r_kind) :: qr,qs,qh,qg,rho,ntr,nts,nth,ntg 
+  REAL(r_kind) :: rainIceRefl_hh,rainIceRefl_vv,rainIceRefl_hv,zdr
+  REAL(r_kind) :: fracqrs,fracqrh,fracqrg
+  REAL(r_kind) :: fracqs,fracqh,fracqg
+  REAL(r_kind) :: fms,fmh,fmg,fws,fwh,fwg,rhoms,rhomh,rhomg
+  REAL(r_kind) :: qrf,qsf,qhf,qgf
+  REAL(r_kind) :: alphaa_ws,alphab_ws,alphaa_wh,alphab_wh,alphaa_wg,alphab_wg
+  REAL(r_kind) :: alphak_ws,alphak_wh,alphak_wg
+  REAL(r_kind) :: rainReflH,ZdrysnowH,ZwetsnowH
+  REAL(r_kind) :: rainReflV,ZdrysnowV,ZwetsnowV
+  REAL(r_kind) :: ZdryhailH,ZwethailH,ZdrygrplH,ZwetgrplH
+  REAL(r_kind) :: ZdryhailV,ZwethailV,ZdrygrplV,ZwetgrplV
+  REAL(r_kind) :: rainReflHV,ZdrysnowHV,ZwetsnowHV
+  REAL(r_kind) :: ZdryhailHV,ZwethailHV,ZdrygrplHV,ZwetgrplHV
+  REAL(r_kind) :: log_ref
+  REAL(r_kind) :: rho_0rs,rho_0rh,rho_0rg,temp
+  REAL(r_kind) :: temH,temV,temHV
+
+  INTEGER :: flg
+
+  REAL(r_kind) :: tair_C
+  REAL(r_kind) :: z_snow_thom
+  REAL(r_double) :: gamma,exp_term,gam_term
```

Comment:
The variable, gamma, has been identified as unused.  Please remove unused variable.

Reply:
We removed it as you mentioned.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R778